src/app/qgslayertreeviewtemporalindicator.cpp

```diff
@@ -0,0 +1,93 @@
+/***************************************************************************
+                         qgslayertreeviewtemporalindicator.cpp
+                         ---------------
+    begin                : February 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslayertreeviewtemporalindicator.h"
+#include "qgslayertreeview.h"
+#include "qgslayertree.h"
+#include "qgslayertreemodel.h"
+#include "qgslayertreeutils.h"
+#include "qgsrasterlayer.h"
+#include "qgisapp.h"
+
+QgsLayerTreeViewTemporalIndicatorProvider::QgsLayerTreeViewTemporalIndicatorProvider( QgsLayerTreeView *view )
+  : QgsLayerTreeViewIndicatorProvider( view )
+{
+}
+
+void QgsLayerTreeViewTemporalIndicatorProvider::connectSignals( QgsMapLayer *layer )
+{
+  if ( !( qobject_cast<QgsVectorLayer *>( layer ) || qobject_cast<QgsRasterLayer *>( layer ) ) )
+    return;
+  QgsMapLayer *mapLayer = layer;
+  connect( mapLayer, &QgsMapLayer::dataSourceChanged, this, &QgsLayerTreeViewTemporalIndicatorProvider::onLayerChanged );
+
+  if ( mapLayer && mapLayer->dataProvider() )
+  {
+    mLayer = mapLayer;
```

Comment:
Don't store the mapLayer here as a member -- this indicator provider will be used for ALL map layers in the project. (i.e. only one instance of the provider is ever created, it's not layer-specific)